listenbrainz/webserver/views/test/test_profile.py

```diff
@@ -94,6 +94,15 @@ def test_delete(self, mock_publish_data_to_queue):
         user = db_user.get(self.user['id'])
         self.assertIsNone(user)
 
+    
+    def test_delete_listens(self):
+        self.temporary_login(self.user['login_id'])
+        r = self.client.get(url_for('profile.delete_listens'))
+        self.assert200(r)
+
+        r = self.client.post(url_for('profile.delete_listens'), data={'token': self.user['auth_token']})
```

Comment:
also, let's add a test that makes sure that the endpoint returns a 4xx if the auth token isn't provided or if the auth token is incorrect.

Reply:
Is it ok if I test against 302 because in case auth token is missing or invalid the user is redirected to profile page. If I explicitly specify 4xx then the page is not automatically redirected.

Reply:
The behaviour seems incorrect then. The code should return we should `raise Unauthorized` if the token is not valid.

Let's fix the actual code so that the test which checks for a 401 Unauthorized status passes

Reply:
Did exactly the same.

Reply:
I have added the unauthorized check. The test for invalid token passed but if I don't pass any token or pass None to it then it gets status 200 in tests but in actual if I delete token it raises unauthorized. Can you please look at th [test code](https://gist.github.com/shivam-kapila/d14dbde481c0208bd95bd87c61eabe7e).

The modified actual code [here](https://gist.github.com/shivam-kapila/246b8b2ac3722a04ee7ea45a8308242a#file-delete_listens-modified-code-L14)

Reply:
In the test, the request response is stored in `r` and the assertion is on `response`

Reply:
Thanks a lot. I have pushed all the tests.